doc/specs/Font features and axes-spec.md

```diff
@@ -0,0 +1,92 @@
+
+### **Axes of variation**
+
+Specifying axes of variation is done in an extremely similar manner to the way font features are specified - a 4-character tag is used to specify which font axis is being modified and a numerical value is provided to specify the value the axis should be set to. For example, {'slnt', 20} specifies that the 'slant' axis should be set to 20.
+
+There is also a standard list of axes of variation, and each axis has its own default. We will approach this the same way we approached font features, by allowing users to specify additional features or omit features without needing to redefine the defaults.
+
+## UI/UX Design
+
+Users will be able to add a new setting to their font objects (added in [#10433](https://github.com/microsoft/terminal/pull/10433)). The resultant font object may look something like this
+
+```
+"font": {
+    "face": "Cascadia Code",
+    "size": 12,
+    "features": {
+        "ss03": 1,
+        "liga": 0
+    },
+    "axes": {
+        "slnt": 20.5
+    }
+}
+```
+
+## Capabilities
+
+### Accessibility
+
+Should not affect accessibility.
+
+### Security
+
+Should not affect security.
+
+### Reliability
+
+Aside from additional parsing required for the settings file (which inherently offers more locations for parsing to fail), we need to be careful about badly formed/non-existant feature tags or axes specified in the user-defined dictionaries. We must make sure to ignore such declarations (perhaps alongside emitting a warning to the user) and only apply those that are correctly formed and exist.
+
+### Compatibility
+
+Older versions of Windows may not have the DWrite updates that allow for defining font features and axes of variation. We must make sure to fallback to the current implementation in these cases.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+We will need to monitor any potential performance issues caused by this feature. Since the extra work being done is only during initialization, the performance issues will probably be negligible if at all, but it will still be wise to check.
+
+## Future considerations
+
+DWrite additionally offers the ability to vary the font features across runs of text. However, for our initial implementation of this feature, we will only apply font features to the entire buffer. If/when we decide to allow specifying font features for particular runs of text, we can lean into our existing mechanisms of splitting up runs of text to implement that.

```

Comment:
This offers us some caching opportunities, no?

Reply:
Not entirely sure what you mean here, what would we cache?

Reply:
yea I don't know what this is referring to

Reply:
I DON'T EITHER OKAY 😁